src/EventEdition/LocationPanel.vala

```diff
@@ -187,6 +192,49 @@ public class Maya.View.EventEdition.LocationPanel : Gtk.Grid {
             debug (error.message);
         }
     }
+
+    private async void find_location (double latitude, double longitude) {
+        if (find_cancellable != null) {
+            find_cancellable.cancel ();
+        }
+        
+        find_cancellable = new GLib.Cancellable ();
+        Geocode.Location location = new Geocode.Location (latitude, longitude);
+        var reverse = new Geocode.Reverse.for_location (location);
+        
+        try {
+            var address = yield reverse.resolve_async (find_cancellable);
+            var builder = new StringBuilder ();
+            if (address.street != null) {
+                builder.append (address.street);
+                if (address.town != null) {
+                    builder.append (", ");
+                    builder.append (address.town);
+                }
+                if (address.county != null) {
+                    builder.append (", ");
+                    builder.append (address.county);
+                }
+                if (address.postal_code != null) {
+                    builder.append (", ");
+                    builder.append (address.postal_code);
+                }
+                if (address.country != null) {
+                    builder.append (", ");
+                    builder.append (address.country);
+                }
+            } else {
+                builder.append (address.name);
+                if (address.country != null) {
+                    builder.append (", ");
+                    builder.append (address.country);
+                }
+            }
+            location_entry.text = builder.str;
```

Comment:
elementary code style:  Should be blank line after end of last if clause (before location_entry.text = ...)